crates/brix_commands/src/command/search_replace.rs

```diff
@@ -0,0 +1,82 @@
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::PathBuf;
+
+use log::info;
+use regex::Regex;
+use validator::Validate;
+
+use crate::command::{Command, ProcessedCommandParams};
+use brix_errors::BrixError;
+
+#[derive(Debug)]
+pub struct SearchReplaceParams {
+    destination: PathBuf,
+    search: String,
+    replace: String,
+}
+
+impl PartialEq for SearchReplaceParams {
+    fn eq(&self, other: &Self) -> bool {
+        return self.destination == other.destination
+            && self.search.as_str() == other.search.as_str()
+            && self.replace == other.replace;
+    }
+}
+
+#[derive(Debug, Validate)]
+struct Params {
+    #[validate(required)]
+    destination: Option<PathBuf>,
+    #[validate(required)]
+    search: Option<String>,
+    #[validate(required)]
+    replace: Option<String>,
+}
+
+pub struct SearchReplaceCommand {}
+
+impl SearchReplaceCommand {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl Command for SearchReplaceCommand {
+    fn run(&self, pcp: ProcessedCommandParams) -> Result<(), BrixError> {
+        let cp = Params {
+            destination: pcp.destination,
+            search: pcp.search,
+            replace: pcp.replace,
+        };
+        cp.validate()?;
+
+        let dest = cp.destination.unwrap();
+        info!("reading to string from '{}'", dest.clone().display());
+        let data = fs::read_to_string(dest.clone()).or_else(|_err| {
```

Comment:
is there not already an IO error that can be wrapped? or... bad idea

Reply:
Yeah but I'd like to map it to a more friendly BrixError in this case but now that I think about it the read could fail due to multiple reasons.... so I should probably include the IO error itself later.

Reply:
right, I'm fine with wrapping with the brix error, but generally you should wrap the original exception too